fixSpec.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+import yaml
+from yaml.loader import FullLoader
+import copy
+from collections import OrderedDict
+
+INFILE = "openapi.yaml"
+OUTFILE = "openapi.fixed.yaml"
+
+# what tags I want to keep
+WANT_TAGS = ["Projects", "Devices", "Organizations"]
+
+# what paths I want to keep
+WANT_PATHS = [
+    '/projects',
+    '/projects/{id}',
+    '/projects/{id}/devices',
+    '/devices/{id}',
+    '/organizations',
+    '/organizations/{id}/projects',
+]
+
+
+def setup_yaml():
+    """ https://stackoverflow.com/a/8661021 """
+
+    def represent_dict_order(self, data): return self.represent_mapping(
+        'tag:yaml.org,2002:map', data.items())
+    yaml.add_representer(OrderedDict, represent_dict_order)
+
+
+setup_yaml()
+
+
+def fixNumInt(di):
+    if isinstance(di, dict):
+        for k, v in di.items():
+            if k == "format" and v == "integer":
+                di["type"] = "integer"
+                di["format"] = "int32"
+            fixNumInt(v)
+    elif isinstance(di, list):
+        for v in di:
+            fixNumInt(v)
+
+
+def fixExplode(di):
```

Comment:
Does the `explode` param cause invalid python to be generated, or is this a stylistic change?

Reply:
it's causing error when running autorest:
```
  File "/home/tomk/.autorest/@autorestpython@6.0.1/node_modules/@autorest/python/autorest/codegen/serializers/parameter_serializer.py", line 52, in serializeparameter
    raise ValueError("Got a explode boolean on a non-array schema")
ValueError: Got a explode boolean on a non-array schema
fatal   | Process() cancelled due to failure 
error   |   Error: Plugin codegen reported failure.
error   | Autorest completed with an error. If you think the error message is unclear, or is a bug, please declare an issues at https://github.com/Azure/autorest/issues with the error message you are seeing.
```

Reply:
Autorest doesn't like if there `explode: true` in primitive types, like

```
      - description: Force the deletion of the device, by detaching any storage volume
          still active.
        explode: true
        in: query
        name: force_delete
        required: false
        schema:
          type: boolean
        style: form
```

.. `explode: true` doesn't really make sense for anything but arrays and objects IMO

Reply:
I made it so that `explode` is kept only for array and object parameters (in [05dc709](https://github.com/t0mk/equinixmetalpy/pull/1/commits/05dc709b3ff82f3450cfa67555d81a1eb7872598))

Reply:
Ah, so we have some `explode: true` specified for primitive types?  You're right, that flag doesn't have meaning for primitive types...we'll have to dig into where that's coming from and why it passes the Metal API spec validation. https://swagger.io/docs/specification/serialization/

Reply:
@ctreatma I think it's nonsensical to have `explode` specified for simple types. I think it passes the validation because in Equninix you use the openapi-generator tools, and they don't mind it.

This project is using azure/autorest, they have their own validation logic which errs on explode on, for example, string.

I will try to get the list of params with `explode` on simple types in the next comment. Maybe we can pass it to the API team.

Thank You for your feeback.

Reply:
I took a closer look at [the spec](https://api.equinix.com/metal/v1/api-docs), and I see `explode: false` for some parameters; the only place I see `explode: true` is on an object.  Is autorest treating `explode: false` as `true`, or is that flag changing somewhere?